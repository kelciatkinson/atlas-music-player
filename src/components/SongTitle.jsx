export default function SongTitle() {
  return <div className="w-100 -mt-68 font-(--font-sans)">
    <p className="h-7 text-2xl font-bold">Painted in Blue</p>
    <p className="pt-2 text-(--color-grey)">Soul Canvas</p>
  </div>
}
