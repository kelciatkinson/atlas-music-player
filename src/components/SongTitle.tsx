import React from "react";


interface SongTitleProps {
  title: string;
  artist: string;
}

export default function SongTitle({ title, artist }: SongTitleProps) {
  return (
    <div className="w-100 -mt-68 font-(--font-sans)">
      <p className="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)">{title}</p>
      <p className="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)">{artist}</p>
    </div>
  );
}
