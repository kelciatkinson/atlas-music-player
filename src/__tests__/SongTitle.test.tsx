import { render } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import SongTitle from "../components/SongTitle";
import React from "react";


describe("SongTitle Snapshot Tests", () => {
  test("renders with title and artist", () => {
    const { container } = render(
      <SongTitle title="Beautiful Song" artist="Amazing Artist" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with empty title", () => {
    const { container } = render(
      <SongTitle title="" artist="Unknown Artist" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with empty artist", () => {
    const { container } = render(
      <SongTitle title="Mystery Song" artist="" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with both empty", () => {
    const { container } = render(
      <SongTitle title="" artist="" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with long title and artist names", () => {
    const { container } = render(
      <SongTitle 
        title="This is a Very Long Song Title That Might Wrap to Multiple Lines"
        artist="This is Also a Very Long Artist Name That Could Be A Little Lengthy"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders with special characters", () => {
    const { container } = render(
      <SongTitle 
        title="Song with Special Characters: !@#$%^&*()"
        artist="Artist & The Band (ft. Someone Else)"
      />
    );
    expect(container).toMatchSnapshot();
  });
});