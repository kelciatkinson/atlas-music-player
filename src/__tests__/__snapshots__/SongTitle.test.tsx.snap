// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Snapshot Tests > renders with both empty 1`] = `
<div>
  <div
    class="w-100 -mt-68 font-(--font-sans)"
  >
    <p
      class="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)"
    />
    <p
      class="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)"
    />
  </div>
</div>
`;

exports[`SongTitle Snapshot Tests > renders with empty artist 1`] = `
<div>
  <div
    class="w-100 -mt-68 font-(--font-sans)"
  >
    <p
      class="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)"
    >
      Mystery Song
    </p>
    <p
      class="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)"
    />
  </div>
</div>
`;

exports[`SongTitle Snapshot Tests > renders with empty title 1`] = `
<div>
  <div
    class="w-100 -mt-68 font-(--font-sans)"
  >
    <p
      class="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)"
    />
    <p
      class="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)"
    >
      Unknown Artist
    </p>
  </div>
</div>
`;

exports[`SongTitle Snapshot Tests > renders with long title and artist names 1`] = `
<div>
  <div
    class="w-100 -mt-68 font-(--font-sans)"
  >
    <p
      class="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)"
    >
      This is a Very Long Song Title That Might Wrap to Multiple Lines
    </p>
    <p
      class="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)"
    >
      This is Also a Very Long Artist Name That Could Be A Little Lengthy
    </p>
  </div>
</div>
`;

exports[`SongTitle Snapshot Tests > renders with special characters 1`] = `
<div>
  <div
    class="w-100 -mt-68 font-(--font-sans)"
  >
    <p
      class="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)"
    >
      Song with Special Characters: !@#$%^&*()
    </p>
    <p
      class="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)"
    >
      Artist & The Band (ft. Someone Else)
    </p>
  </div>
</div>
`;

exports[`SongTitle Snapshot Tests > renders with title and artist 1`] = `
<div>
  <div
    class="w-100 -mt-68 font-(--font-sans)"
  >
    <p
      class="h-7 text-2xl font-bold text-(--color-text) dark:text-(--color-dark-text)"
    >
      Beautiful Song
    </p>
    <p
      class="pt-2 text-(--color-secondary) dark:text-(--color-dark-secondary)"
    >
      Amazing Artist
    </p>
  </div>
</div>
`;
